client/app/components/viz/VariantInspectCard.vue

```diff
@@ -2434,6 +2448,16 @@ export default {
           self.loadData();
       })
     },
+
+    mosaicVariant: {
+      handler(newMosaicVariant, oldMosaicVariant) {
+        if (newMosaicVariant !== oldMosaicVariant) {
```

Comment:
Why do we need to test of the newMosaicVariant doesn't equal the old one? Isn't a watch method on 'mosaicVariant' only called when it has changed?

Reply:
I put the 'if statement' here because of Avoiding Unwanted Side Effects: If mosaicVariant is an object or array and the reference to it doesn't change but its properties or elements change, the watcher would trigger even if the overall reference hasn't changed. The if check ensures that you only perform the necessary action if the content has actually changed.